includes/class-edd-discount.php

```diff
@@ -244,13 +228,93 @@ public function __construct( $_id_or_code_or_name = false, $by_code = false, $by
 	 * @return mixed
 	 */
 	public function __get( $key ) {
+
 		$key = sanitize_key( $key );
 
-		if ( method_exists( $this, 'get_' . $key ) ) {
+		if( 'discount_id' === $key || 'ID' == $key ) {
+			return (int) $this->id;
+		} else if ( method_exists( $this, 'get_' . $key ) ) {
 			return call_user_func( array( $this, 'get_' . $key ) );
 		} else if ( property_exists( $this, $key ) ) {
 			return $this->{$key};
 		} else {
+
+			switch( $key ) {
+
+				// Account for old property keys from pre 3.0
+				case 'post_author' :
+					break;
+				case 'post_date' :
+				case 'post_date_gmt' :
+
+					return $this->date_created;
+					break;
+
+				case 'post_content' :
+				case 'post_title' :
+
+					return $this->name;
+					break;
+
+				case 'post_excerpt' :
+				case 'post_status' :
+					return $this->status;
+					break;
+
+				case 'comment_status' :
+				case 'ping_status' :
+				case 'post_password' :
+				case 'post_name' :
+				case 'to_ping' :
+				case 'pinged' :
+				case 'post_modified' :
+				case 'post_modified_gmt' :
+				case 'post_content_filtered' :
+				case 'post_parent' :
+				case 'guid' :
+				case 'menu_order' :
+				case 'post_mime_type' :
+				case 'comment_count' :
+				case 'filter' :
+				case 'post_type' :
```

Comment:
I think we should return `edd_discount` for `post_type` to help ensure any custom integrations that check for the post type still function.